plugins/Marketplace/PluginTrial/Service.php

```diff
@@ -0,0 +1,137 @@
+<?php
+
+/**
+ * Matomo - free/libre analytics platform
+ *
+ * @link https://matomo.org
+ * @license http://www.gnu.org/licenses/gpl-3.0.html GPL v3 or later
+ */
+
+namespace Piwik\Plugins\Marketplace\PluginTrial;
+
+use Exception;
+use Piwik\Config\GeneralConfig;
+use Piwik\Piwik;
+use Piwik\Session;
+
+final class Service
+{
+    public function __construct()
+    {
+    }
+
+    /**
+     * Creates a trial request (and sends a mail to all super users)
+     *
+     * @return void
+     */
+    public function request(string $pluginName): void
+    {
+        if (!$this->isEnabled()) {
+            return;
+        }
+
+        $request = new Request($pluginName, new Storage($pluginName));
```

Comment:
We need to update the storage in some way to store the display name of the plugin.

I have updated the frontend part already, but the Email and Notification should also use it.